loopin.go

```diff
@@ -209,6 +247,63 @@ func newLoopInSwap(globalCtx context.Context, cfg *swapConfig,
 	}, nil
 }
 
+// awaitProbe waits for a probe payment to arrive and cancels it. This is a
+// workaround for the current lack of multi-path probing.
+func awaitProbe(ctx context.Context, lnd lndclient.LndServices,
+	probeHash lntypes.Hash) (chan error, error) {
+
+	// Subscribe to the probe invoice.
+	updateChan, errChan, err := lnd.Invoices.SubscribeSingleInvoice(
+		ctx, probeHash,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait in the background for the probe to arrive.
+	probeResult := make(chan error, 1)
+
+	go func() {
```

Comment:
nit:  you could reduce the nesting a bit by extracting this into a closure and go running that separately.

Reply:
Doesn't a closure result in exactly the same amount of nesting?

Reply:
If I'm not mistaken you can still get rid of the `for` since we return from every case of the select.

Reply:
Ok, discussed this offline, the for loop is required indeed.